File: benchmark/sum/bench_sum.c
#include <stdio.h>
#include <time.h>

int main(void) {
    struct timespec s, e;
    clock_gettime(CLOCK_MONOTONIC, &s);
    int sum = 0;
    for (int i = 0; i <= 100000000; i++) sum += i;
    clock_gettime(CLOCK_MONOTONIC, &e);
    double ms = (e.tv_sec-s.tv_sec)*1000.0 + (e.tv_nsec-s.tv_nsec)/1e6;
    printf("result: %d  time: %.2fms\n", sum, ms);
    return 0;
}

File: benchmark/sum/bench_sum.rs
use std::time::Instant;

fn main() {
    let start = Instant::now();
    let mut sum: i32 = 0;
    let mut i: i32 = 0;
    while i <= 100_000_000 { sum = sum.wrapping_add(i); i += 1; }
    let ms = start.elapsed().as_secs_f64() * 1000.0;
    println!("result: {}  time: {:.2}ms", sum, ms);
}
